### PR TITLE
deps(ci): use v0.17.0 of `cargo semver-checks`

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by jxs — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@obi1kenobi It seems like v0.17.0 unfortunately isn't without problems :(

**Comment by obi1kenobi** [comment]
Well, this is exciting! More test cases for the integration suite :)

Thanks for the report, apologies for the inconvenience.

This is the new number 1 on my queue for today.

**Comment by obi1kenobi** [comment]
Amazing, 22 test crates in `trustfall-rustdoc-adapter` and 4 test crates in `cargo-semver-checks` specifically for testing re-exports edge cases, and I still failed to come up with and cover the edge case present in these two crates.

Thankfully, a defensive assert caught the problem so it was easy to track down. It happens when a re-export relies on defaults in generic parameters to omit them in its signature.

It won't be a difficult fix, I expect to have it published within an hour or two.

**Comment by thomaseizinger** [comment]
> It happens when a re-export relies on defaults in generic parameters to omit them in its signature.

Haha, wow. That is pretty niche I guess!

Who would have thought that our repo is such a good testground!

**Comment by obi1kenobi** [comment]
> Who would have thought that our repo is such a good testground!

Your repo is absolutely dominating our issues-discovered hall of fame right now, with 4 entries in the regression suite. Bevy and [the Rust bindings for Core Foundation](https://github.com/servo/core-foundation-rs) share second place with 2 issues apiece.

v0.17.1 is now published and includes the fix for this issue. Thanks for bearing with us — we're thrilled you're using cargo-semver-checks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
       - name: Check public API for semver violations
         if: steps.check-released.outputs.code == 200 # Workaround until https://github.com/obi1kenobi/cargo-semver-check/issues/146 is shipped.
         run: |
-          wget -q -O- https://github.com/obi1kenobi/cargo-semver-checks/releases/download/v0.16.2/cargo-semver-checks-x86_64-unknown-linux-gnu.tar.gz | tar -xz -C ~/.cargo/bin
+          wget -q -O- https://github.com/obi1kenobi/cargo-semver-checks/releases/download/v0.17.0/cargo-semver-checks-x86_64-unknown-linux-gnu.tar.gz | tar -xz -C ~/.cargo/bin
```
Comment (obi1kenobi) [inline .github/workflows/ci.yml]:
```suggestion
          wget -q -O- https://github.com/obi1kenobi/cargo-semver-checks/releases/download/v0.17.1/cargo-semver-checks-x86_64-unknown-linux-gnu.tar.gz | tar -xz -C ~/.cargo/bin
```

**Comment by thomaseizinger** [comment]
> v0.17.1 is now published and includes the fix for this issue. Thanks for bearing with us — we're thrilled you're using cargo-semver-checks!

`0.17.1` looks solid!
